std/os/linux/index.zig

```diff
@@ -1254,6 +1268,21 @@ pub fn timerfd_settime(fd: i32, flags: u32, new_value: *const itimerspec, old_va
     return syscall4(SYS_timerfd_settime, @intCast(usize, fd), @intCast(usize, flags), @ptrToInt(new_value), @ptrToInt(old_value));
 }
 
+/// fcntl with zero parameters
+pub fn fcntl0(fildes: i32, cmd: i32) usize {
+    return syscall2(SYS_fcntl, varToSyscall(fildes), varToSyscall(cmd));
+}
+
+/// fcntl with one parameter
+pub fn fcntl1(fildes: i32, cmd: i32, arg: var) usize {
```

Comment:
Instead of `var` this should just be `usize`, and the caller has to figure out how their argument should become a `usize`. The syscall needs usizes, and typically the logic for figuring out how to do that is done in this file, however if the type is not known, that logic needs to be done in the caller. Likewise for the darwin equivalents.

Reply:
This is the one that I am having trouble with. If it's a pointer to a struct, then is usize guaranteed to be the size of a pointer? On darwin, we're not yet calling directly to a syscall/asm, so that's why I went with var + varToSyscall -- to get as close as possible to the final underlying declaration that zig is aware of. `fcntl` on darwin is a c function with va_args

Reply:
I am also confused, other function calls in `os/linux/index.zig` follow their `man` page equivalent. `fcntl` allows for va_args so, isn't that var? maybe I should recreate the function decl as `pub fn fcntl(fildes: i32, cmd: i32, args: ...) usize` and then call the appropriate syscall inside that function?

*for reference:*
```zig
pub fn epoll_ctl(epoll_fd: i32, op: u32, fd: i32, ev: *epoll_event) usize {
    return syscall4(SYS_epoll_ctl, @intCast(usize, epoll_fd), @intCast(usize, op), @intCast(usize, fd), @ptrToInt(ev));
}
```

Reply:
> If it's a pointer to a struct, then is usize guaranteed to be the size of a pointer?

Yes, that's guaranteed by the Linux ABI and the Darwin ABI.

In MacOS-specific code, we should probably call the `fnctl` C function directly. This will follow the C ABI for var args and will be correct.

For Linux-specific code, I think the `epoll_ctl` example is good. Have a function with correct types and then all it does is make the syscall.

For code that should be shared by both, a function can abstract over the differences for one particular `F_CMD`.

Reply:
I don't think we need to use `var` for anything to do with `fnctl`.